### DISCUSSION TITLE
Can flux notifications for commit status be from different repositories

### DISCUSSION THREAD

**Discussion by timm088** [fluxcd/flux2#1930] [Q&A]
Hi,

Somewhat new to using flux v2 so hoping someone may be able to help

* We have setup a notification controller, secrets, provider and all of this is working for github commit status
* However, we see an error , like the below from the notification controller

`"error":"could not create commit status: POST https://api.github.com/repos/(our-org)/(repository)/statuses/commit-hash: 422 No commit found for SHA: commit-hash`

In my example i've change things like 'commit-hash' , but hopefully describes the issue

The commit hash it sends to the 'address' repository in the post, is the commit hash of the 'infrastructure' repository, which, in our case is a separate repository from the application repository. The error is 'correct', that hash doesn't exist in the application repository, because its from the 'infrastructure' one.

* ? - Is there a way to use the setup we have (separate app deployment overrides to the 'source' image repository'). 
* Or, should use have our 'deployment' manifests living with the applications themselves?

**Comment by stefanprodan** [comment]
In the Alert definition you should point Flux to the infrastructure repository and set the infrastructure Kustomization as source.

**Reply by timm088** [reply]
Hey @stefanprodan thanks for the tip. Just so i understand / can communicate clearly

* Application A - github.com/org/repoa - this is where the docker image is built
* Infrastructure a - github.com/org/infraa - this is where the depployment kustomization is (e.g., replica count, Alert definition etc)

I'm not quite following how pointing flux to the infra repo and infra kustomization would help?

**Reply by timm088** [reply]
Any further details here by chance? Still can't figure out what you meant by the last comment. We may have this arranged incorrectly, but am hoping that its just a matter of configuration missing.

**Reply by stefanprodan** [reply]
If you want to get commit status updates on the app repo then the deployment manifests should come from there. The commit sha that Flux knows about, is the commit coming from the repo that Flux syncs.

**Reply by sunilpandey1989** [reply]
@timm088 I am also facing same issue. Did you get any resolution to the issue ?

**Reply by eguarc** [reply]
> @timm088 I am also facing same issue. Did you get any resolution to the issue ?

probably here's the issue you're facing:

https://v2-0.docs.fluxcd.io/flux/guides/notifications/

> Limitations
> The git notification providers require that a commit hash present in the meta data of the event. Therefore the providers will only work with Kustomization as an event source, as it is the only resource which includes this data.
